dodiscover/metrics.py

```diff
@@ -146,3 +145,29 @@ def structure_hamming_dist(
         diff = diff + diff.T
         diff[diff > 1] = 1  # Ignoring the double edges.
         return np.sum(diff) / 2
+
+
+def toporder_divergence_arr(A: NDArray, order: List[int]):
+    """Compute topological ordering divergence.
+
+    Given an adjacency matrix A and a topological ordering on its entries,
+    the topological ordering divergence (Rolland et al. (2022)) counts
+    the number of edges of A that can not be recovered due to the choice of
+    the topological ordering.
+
+    Parameters
+    ----------
+    A : NDArray
+        Input adjacency matrix representation of a directed acyclic graphs.
+    order : List[int]
+        A topological ordering on the nodes of the graph.
+
+    Return
+    ------
+    err : int
+        Sum of the number of edges of A not admitted by the given order.
+    """
+    err = 0
+    for i in range(len(order)):
+        err += A[order[i + 1 :], order[i]].sum()
```

Comment:
Suggest more descriptive names for people who are new to the idea

Reply:
maybe something like:

```
from numpy.typing import NDArray
from typing import List

def toporder_divergence_arr(A: NDArray, order: List[int]) -> int:
    if len(order) != A.shape[0] or A.shape[0] != A.shape[1]:
        raise ValueError("The dimensions of the adjacency matrix and the order list do not match.")
        
    total_edges_not_admitted = 0
    for i in range(len(order)):
        total_edges_not_admitted += A[order[i + 1 :], order[i]].sum()
    return total_edges_not_admitted

```

Reply:
I rewrote a bit the docstring explaining that the error introduced by the topological ordering is a lower bound on the false nagatives (i.e. if order does not allow to recover an edge, this will for sure count as false negative in structural hamming distance), and went for 
```
def toporder_divergence_arr(A: NDArray, order: List[int]) -> int:
    if len(order) != A.shape[0] or A.shape[0] != A.shape[1]:
        raise ValueError("The dimensions of the adjacency matrix and the order list do not match.")
        
    false_negatives_from_order = 0
    for i in range(len(order)):
        false_negatives_from_order += A[order[i + 1 :], order[i]].sum()
    return false_negatives_from_order
```